sdk/scheduler/src/main/java/org/apache/mesos/scheduler/plan/Step.java

```diff
@@ -26,17 +26,25 @@
      * @see {@link #updateOfferStatus(Collection<Offer.Operation>)} which returns the outcome of the
      *      {@link OfferRequirement}
      */
-    Optional<OfferRequirement> start();
+    Optional<PodInstanceRequirement> start();
 
     /**
      * Notifies the Step whether the {@link OfferRequirement} previously returned by
-     * {@link #start()} has been successfully accepted/fulfilled. The {@code operations} param is
+     * {@link #getOfferRequirement()} has been successfully accepted/fulfilled. The {@code operations} param is
      * empty when no offers matching the requirement previously returned by {@link #clone()
-     * could be found. This is only called if {@link #start()} returned a non-{@code null}
+     * could be found. This is only called if {@link #getOfferRequirement()} returned a non-{@code null}
      * {@link OfferRequirement}.
      */
     void updateOfferStatus(Collection<Offer.Operation> operations);
 
+    Optional<String> getAsset();
+
+    /**
+     * Provides the assets which the Step is currently operating on to avoid contention.
+     * @return
```

Comment:
Also what's the relationship between this and `getAsset()`? Would they both return the same asset name? Could it instead be `getAsset()` + `isAssetDirty()`?